tests/integration/test_nft_creation.py

```diff
@@ -0,0 +1,418 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import pytest
+from ocean_lib.models.v4.erc20_token import ERC20Token
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.erc721_token import ERC721Token
+from ocean_lib.models.v4.models_structures import ErcCreateData
+from ocean_lib.utils.addresses_utils import (
+    get_nft_factory_address,
+    get_nft_template_address,
+)
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+from ocean_lib.web3_internal.contract_utils import get_contracts_addresses
+from tests.resources.helper_functions import (
+    get_another_consumer_wallet,
+    get_consumer_wallet,
+    get_publisher_wallet,
+)
+from web3 import exceptions
+from ocean_lib.web3_internal.wallet import Wallet
+from web3 import Web3
+
+
+def deploy_erc721_erc20(web3, config, erc721_publisher: Wallet, erc20_minter: Wallet):
+    """Helper function to deploy an ERC721Token using erc721_publisher Wallet
```

Comment:
For better reusabiliity, please move this into a tests/helpers file or something similar. We will probably reuse this function or similar ones, so let's keep it separate.